dc/s2n-quic-dc/src/stream/tests/shared_cache.rs

```diff
@@ -189,3 +213,246 @@ async fn test_connection(
+            tokio::time::error::Elapsed { .. }
+        ));
+
+        clients.push(client_stream);
+    }
+
+    let mut servers = Vec::new();
+    for _ in 0..stream_count {
+        let (stream, _addr) = app_server.accept().await.unwrap();
+        servers.push(stream);
+    }
+
+    let test_message = b"Hello from server!";
+    for mut stream in servers {
+        let mut message_slice = &test_message[..];
+        stream.write_from(&mut message_slice).await.unwrap();
+    }
+
+    for mut stream in clients {
+        let mut buffer: Vec<u8> = Vec::new();
+        let bytes_read = stream.read_into(&mut buffer).await.unwrap();
+        assert_eq!(
+            &buffer[..bytes_read],
+            test_message,
+            "Client should receive the correct message"
+        );
+    }
+}
+
+#[cfg(not(target_os = "macos"))]
+#[tokio::test]
+async fn test_kernel_queue_full_application_crash() {
+    init_tracing();
+    let test_event_subscriber = NoopSubscriber {};
+    let unix_socket_path = PathBuf::from("/tmp/kernel_queue_crash.sock");
+
+    let (stream_client, _) = create_stream_client();
+    let handshake_server = create_handshake_server().await;
+
+    let handshake_addr = handshake_server.local_addr();
+    stream_client
+        .handshake_with(handshake_addr, server_name())
+        .await
+        .unwrap();
+    info!("Handshake completed");
+
+    let manager_server = manager::Server::<ServerProvider, NoopSubscriber>::builder()
+        .with_address("127.0.0.1:0".parse().unwrap())
+        .with_protocol(Protocol::Tcp)
+        .with_udp(false)
+        .with_workers(NonZeroUsize::new(1).unwrap())
+        .with_socket_path(&unix_socket_path)
+        .with_backlog(NonZero::new(5000).unwrap())
+        .build(handshake_server.clone(), test_event_subscriber.clone())
+        .unwrap();
+
+    info!(
+        "Manager server created at: {:?}",
+        manager_server.acceptor_addr()
+    );
+    let acceptor_addr = manager_server.acceptor_addr().unwrap();
+
+    let app_server = create_application_server(&unix_socket_path, test_event_subscriber);
+
+    let mut clients = Vec::new();
+    let stream_count = 5000;
+
+    for _ in 0..stream_count {
+        let mut client_stream = stream_client
+            .connect(handshake_addr, acceptor_addr, server_name())
+            .await
+            .unwrap();
+
+        let mut buffer: Vec<u8> = Vec::new();
+        let read_result = tokio::time::timeout(
+            Duration::from_millis(5),
+            client_stream.read_into(&mut buffer),
+        )
+        .await;
+        assert!(read_result.is_err());
+        assert!(matches!(
+            read_result.unwrap_err(),
+            tokio::time::error::Elapsed { .. }
+        ));
+        clients.push(client_stream);
+    }
+
+    drop(app_server);
+
+    for mut stream in clients {
+        let mut buffer: Vec<u8> = Vec::new();
+        let read_result = stream.read_into(&mut buffer).await;
+        assert!(read_result.is_err());
```

Comment:
```suggestion
```

Here and elsewhere -- if you're going to unwrap_err, you generally want just that. This assertion is (a) redundant and (b) hides the Ok(...) result. In this case the Ok result is probably not very interesting, but it's still better if we avoid the antipattern.